### PR TITLE
Feature/import sim batch for hlr pre activated msisdns

### PR DESCRIPTION
**Pull request by la3lma**
Unit and integration tests added for import script that is now modified to set the hss state from a query parameter.

### CODE REVIEW

File: build-all.sh [hunk 1]
```diff
@@ -93,7 +93,7 @@ DIRS_THAT_NEEDS_SERVICE_ACCOUNT_CONFIGS=" \
   bq-metrics-extractor/config \
   auth-server/config prime/config"
 
-SERVICE_ACCOUNT_MD5="b8d22f87b55431ebfa4b68c16d7dc037"
+SERVICE_ACCOUNT_MD5="c54b903790340dd9365fa59fce3ad8e2"
```
Comment (la3lma) [inline build-all.sh]: This is an unrelated change.  It's for making the "build-all.sh" script recognize the latest prime-service-account.json instancde.

File: sim-administration/simmanager/src/integration-test/kotlin/org/ostelco/simcards/admin/SimAdministrationTest.kt [hunk 2]
```diff
@@ -35,6 +44,11 @@ class SimAdministrationTest {
         private lateinit var jdbi: Jdbi
         private lateinit var client: Client
 
+        // ICCID of rirst SIM in sample-sim-batch.csv
```
Comment (prasanthu) [inline sim-administration/simmanager/src/integration-test/kotlin/org/ostelco/simcards/admin/SimAdministrationTest.kt]: spelling mistake `rirst`

File: sim-administration/simmanager/src/main/kotlin/org/ostelco/simcards/admin/ArrowUtils.kt [hunk 3]
```diff
@@ -5,6 +5,5 @@ import arrow.core.Either
 
 /**
  * TODO: 1. Document this method
- *       2. Move it into a convenience libary for Arrow utilities.
  */
 fun <L,R, R2> Either<L, R>.mapRight(f:(R)->R2): Either<L, R2> = this.map(f)
```
Comment (prasanthu) [inline sim-administration/simmanager/src/main/kotlin/org/ostelco/simcards/admin/ArrowUtils.kt]: No newline

File: sim-administration/simmanager/src/main/kotlin/org/ostelco/simcards/inventory/SimInventoryResource.kt [hunk 4]
```diff
@@ -111,15 +112,21 @@ class SimInventoryResource(private val api: SimInventoryApi) {
...
-                            {
-                                error("Failed to upload batch with SIM profiles for HSS ${hss} and SIM profile vendor ${simVendor}",
-                                        ApiErrorCode.FAILED_TO_IMPORT_BATCH, it)
-                            },
-                            { Response.status(Response.Status.OK).entity(asJson(it)) }
-                    ).build()
+            @QueryParam("initialHssState") initialHssStateArg: HssState?,
+            csvInputStream: InputStream): Response {
+
+        val initialHssState = if (initialHssStateArg==null) HssState.NOT_ACTIVATED else initialHssStateArg
```
Comment (kmmatwork) [inline sim-administration/simmanager/src/main/kotlin/org/ostelco/simcards/inventory/SimInventoryResource.kt]: Since this is at input I guess that it could be an idea to check whether the `initialHssStateArg` value, if set, is a valid value? That is matches one of the entries in the `HssState` enum.

File: sim-administration/simmanager/src/integration-test/kotlin/org/ostelco/simcards/admin/SimAdministrationTest.kt (hunk 2 above)
Comment (la3lma) [reply sim-administration/simmanager/src/integration-test/kotlin/org/ostelco/simcards/admin/SimAdministrationTest.kt]: fixed

File: sim-administration/simmanager/src/main/kotlin/org/ostelco/simcards/inventory/SimInventoryResource.kt (hunk 4 above)
Comment (la3lma) [reply sim-administration/simmanager/src/main/kotlin/org/ostelco/simcards/inventory/SimInventoryResource.kt]: It's implicitly checked by the input filter for the applet.   If it's set it will automatically be translated to a  legal value otherwise a HTTP 500 will be thrown (will add explicit test for that).  I don't know if it's possible to test for illegal query parameters to be set or if they fail silently.  I will check.  It would be better to fail than to fail silently, because silent failure of this kind will actually import the data but with the wrong value for initial hss state set.  Good catch.

Comment (la3lma) [reply sim-administration/simmanager/src/main/kotlin/org/ostelco/simcards/inventory/SimInventoryResource.kt]: fixed now.

File: sim-administration/simmanager/src/main/kotlin/org/ostelco/simcards/admin/ArrowUtils.kt (hunk 3 above)
Comment (la3lma) [reply sim-administration/simmanager/src/main/kotlin/org/ostelco/simcards/admin/ArrowUtils.kt]: Fixed
